Add translatewiki link

333nischay opened the pull request:
Ticket: T241500

Query: The footer of SVG Translate includes useful links but lacks a direct link to [Translatewiki](https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate) making it difficult for users to find and contribute translations. Adding this link would improve accessibility and streamline the translation process.

This PR enhances the footer of SVG Translate by adding a direct link to [Translatewiki](https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate), making it easier for users to contribute translations and support the tool.  

 Changes Made:  
1. Added a new translation key:
   - Introduced `"translatewiki": "Translate on Translatewiki"` to the all the language files available in `i18n` directory in the respective languages.  
   - Added corresponding translations in multiple language files to encourage users to help translate the tool.  

2. Updated the footer template:  
   - Modified `templates/base.html.twig` to include the [Translatewiki](https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate) link:  
     ```
     <li><a href="https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate">{{ msg('translatewiki') }}</a></li>
     ```  

This update improves accessibility and streamlines the translation process for contributors.

samwilson commented on templates/base.html.twig:
```diff
@@ -74,6 +74,7 @@
                     {% endset %}
                     <li><a href="https://phabricator.wikimedia.org/maniphest/task/edit/form/1/?title=%5BBUG%5D%20...&description={{ issue_template|escape('url') }}&projects=community-tech,svg_translate_tool&priority=triage">{{ msg('report-issue') }}</a></li>
                     <li><a href="https://meta.wikimedia.org/wiki/Special:MyLanguage/Community_Tech/SVG_translation">{{ msg('help') }}</a></li>
+                    <li><a href="https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate">{{ msg('translatewiki') }}</a></li>
```
This URL should probably just be `https://translatewiki.net/wiki/Special:Translate/svgtranslate`, so that the user's own settings are used for the target language (i.e. setting `language=en` just results in a warning because it's the same as the base language).

samwilson commented on i18n/en.json:
```diff
@@ -84,5 +84,6 @@
 	"report-issue": "Report a bug",
-	"view-source": "View on Github"
+	"view-source": "View on Github",
+	"translatewiki": "Help translate this tool on Translatewiki"
```
I think a shorter message might be good, e.g. "Translate on translatewiki.net", so it doesn't take up too much space:

![image](https://github.com/user-attachments/assets/6c48e95e-3f80-4ba8-8598-aa07141b9106)

Note also that normal way that translatewiki.net is written is like that, all lowercase and with the ".net".

samwilson reviewed (changes requested):
Thanks for this!

How did you create the translations for all the different languages? If you speak all these then that's fine, but if you used machine translation then I'm sorry to say that it's going to be best to remove them.

When adding new messages to a project, we only have to add them to the `en.json` and `qqq.json` files, and translatewiki.net will do the rest for us. They have built-in machine translation and other suggestion systems that make it quick and easy for translators there to add the missing languages, so it's actually better for us to not do them.

333nischay commented:
Sure will do the changes. Thanks for attention.

On Mon, 6 Jan, 2025, 5:47 am Sam Wilson, ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Thanks for this!
>
> How did you create the translations for all the different languages? If
> you speak all these then that's fine, but if you used machine translation
> then I'm sorry to say that it's going to be best to remove them.
>
> When adding new messages to a project, we only have to add them to the
> en.json and qqq.json files, and translatewiki.net will do the rest for
> us. They have built-in machine translation and other suggestion systems
> that make it quick and easy for translators there to add the missing
> languages, so it's actually better for us to not do them.
> ------------------------------
>
> In templates/base.html.twig
> <https://github.com/wikimedia/svgtranslate/pull/768#discussion_r1903376962>
> :
>
> > @@ -74,6 +74,7 @@
>                      {% endset %}
>                      <li><a href="https://phabricator.wikimedia.org/maniphest/task/edit/form/1/?title=%5BBUG%5D%20...&description={{ <https://phabricator.wikimedia.org/maniphest/task/edit/form/1/?title=%5BBUG%5D%20...&description=%7B%7B> issue_template|escape('url') }}&projects=community-tech,svg_translate_tool&priority=triage">{{ msg('report-issue') }}</a></li>
>                      <li><a href="https://meta.wikimedia.org/wiki/Special:MyLanguage/Community_Tech/SVG_translation">{{ msg('help') }}</a></li>
> +                    <li><a href="https://translatewiki.net/wiki/Special:Translate/svgtranslate?group=svgtranslate&language=en&filter=%21translated&action=translate">{{ msg('translatewiki') }}</a></li>
>
> This URL should probably just be
> https://translatewiki.net/wiki/Special:Translate/svgtranslate, so that
> the user's own settings are used for the target language (i.e. setting
> language=en just results in a warning because it's the same as the base
> language).
> ------------------------------
>
> In i18n/en.json
> <https://github.com/wikimedia/svgtranslate/pull/768#discussion_r1903378305>
> :
>
> > @@ -84,5 +84,6 @@
>  	"powered-by-toolforge": "Powered by Toolforge",
>  	"privacy-policy": "Privacy policy",
>  	"report-issue": "Report a bug",
> -	"view-source": "View on Github"
> +	"view-source": "View on Github",
> +	"translatewiki": "Help translate this tool on Translatewiki"
>
> I think a shorter message might be good, e.g. "Translate on
> translatewiki.net", so it doesn't take up too much space:
>
> image.png (view on web)
> <https://github.com/user-attachments/assets/6c48e95e-3f80-4ba8-8598-aa07141b9106>
>
> Note also that normal way that translatewiki.net is written is like that,
> all lowercase and with the ".net".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wikimedia/svgtranslate/pull/768#pullrequestreview-2531003097>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJUFH4STBVGATTXFMGCG3ET2JHDQZAVCNFSM6AAAAABUUNDOFGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMZRGAYDGMBZG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

333nischay commented:
Removed the translatewiki message form all the language files except `en.json` and `qqq.json`. Changed the message to `"translatewiki": "Translate this tool"` and changed the link address to [translatewiki.net](https://translatewiki.net/wiki/Special:Translate/svgtranslate).

samwilson commented on i18n/qqq.json:
```diff
@@ -87,5 +87,7 @@
 	"report-issue": "Label for issue-reporting link to Phabricator. Displayed in the footer on every page.",
-	"view-source": "Label for the link to the Github project page. Displayed in the footer on every page."
+	"view-source": "Label for the link to the Github project page. Displayed in the footer on every page.",
+	"translatewiki": "Translate this tool. Displayed in the footer on every page."
```
These are the instructions for translators, who can already see the "Translate this tool" base text. It's good to explain how the text is to be used (i.e. that it's a link).

samwilson reviewed (changes requested):
Thanks!

It looks like you've still got translations for anp, ar, as, ast, az, and da, as well as introducing some blank lines for other langs.

333nischay commented:
removed the translations form anp, ar, as, ast, az, and da and extra blank lines. Also explained the tool in `qqq.json` as `"translatewiki": "This is a link for the translate tool (translatewiki.net). Displayed in the footer on every page."`

samwilson reviewed (approved):
Looks good, thanks!

You have removed the trailing newlines on all the translation files, but that doesn't matter too much. Let's merge this!
